src/constraints/scalaraffine.jl

```diff
@@ -196,18 +196,65 @@ function MOI.modify!(model::LinQuadOptimizer, index::LCI{S}, change::MOI.ScalarC
     change_matrix_coefficient!(model, row, column, change.new_coefficient)
 end
 
+function _replace_with_matching_sparsity!(model::LinQuadOptimizer, previous::Linear, replacement::Linear, row)
+    rows = fill(row, length(replacement.terms))
+    cols = [model.variable_mapping[t.variable_index] for t in replacement.terms]
+    coefs = MOIU.coefficient.(replacement.terms)
+    change_matrix_coefficients!(model, rows, cols, coefs)
+end
+
+function _replace_with_different_sparsity!(model::LinQuadOptimizer, previous::Linear, replacement::Linear, row)
+    # First, zero out the old constraint function terms
+    rows = fill(row, length(previous.terms))
+    cols = [model.variable_mapping[t.variable_index] for t in previous.terms]
+    coefs = fill(0.0, length(previous.terms))
+    change_matrix_coefficients!(model, rows, cols, coefs)
+
+    # Next, set the new constraint function terms
+    rows = fill(row, length(replacement.terms))
+    cols = [model.variable_mapping[t.variable_index] for t in replacement.terms]
+    coefs = MOIU.coefficient.(replacement.terms)
+    change_matrix_coefficients!(model, rows, cols, coefs)
```

Comment:
Do these account for duplicate coefficients? (Edit: looks from below that the inputs are canonicalized. All the more reason for docstrings.)